app/src/main/java/com/github/se/cyrcle/ui/profile/ProfileScreen.kt

```diff
@@ -0,0 +1,192 @@
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.unit.dp
+import coil.compose.rememberAsyncImagePainter
+import coil.request.ImageRequest
+import coil.transform.CircleCropTransformation
+import com.github.se.cyrcle.model.user.User
+import com.github.se.cyrcle.ui.theme.ColorLevel
+import com.github.se.cyrcle.ui.theme.getButtonColors
+
+@Composable
+fun ProfileScreen(user: User) {
+  var isEditing by remember { mutableStateOf(false) }
+  var firstName by remember { mutableStateOf(TextFieldValue(user.firstName)) }
+  var lastName by remember { mutableStateOf(TextFieldValue(user.lastName)) }
+  var username by remember { mutableStateOf(TextFieldValue(user.username)) }
+
+  var favoriteParkings by remember { mutableStateOf(user.favoriteParkings.toMutableList()) }
+  val parkingFavorites = remember {
+    mutableStateListOf(*List(user.favoriteParkings.size) { true }.toTypedArray())
+  }
+
+  var profilePictureUrl by remember { mutableStateOf(user.profilePictureUrl) }
+  val defaultImageUrl =
+      "https://th.bing.com/th/id/R.aee6adef085a77dfa4708f3fd4a1ffb5?rik=nj%2buNqgLIJU0JQ&pid=ImgRaw&r=0"
+  val filledStarImageUrl =
+      "https://th.bing.com/th/id/R.f493b7a899f0c38194622d00ca70a7ff?rik=MvAc4%2fM5MeISxA&pid=ImgRaw&r=0"
+  val emptyStarImageUrl =
+      "https://cdn0.iconfinder.com/data/icons/thin-voting-awards/57/thin-232_star_favorite_add-1024.png"
+
+  val context = LocalContext.current
+  val imagePickerLauncher =
+      rememberLauncherForActivityResult(ActivityResultContracts.GetContent()) { uri ->
+        uri?.let { profilePictureUrl = it.toString() }
+      }
+
+  DisposableEffect(Unit) {
+    onDispose {
+      favoriteParkings =
+          favoriteParkings.filterIndexed { index, _ -> parkingFavorites[index] }.toMutableList()
+    }
+  }
+
+  Scaffold { innerPadding ->
+    Column(
+        modifier = Modifier.fillMaxSize().padding(innerPadding).padding(16.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(16.dp)) {
+          if (isEditing) {
+            TextField(
+                value = firstName,
+                onValueChange = { firstName = it },
+                label = { Text("First Name") },
+                modifier = Modifier.testTag("FirstNameField"))
+            TextField(
+                value = lastName,
+                onValueChange = { lastName = it },
+                label = { Text("Last Name") },
+                modifier = Modifier.testTag("LastNameField"))
+            TextField(
+                value = username,
+                onValueChange = { username = it },
+                label = { Text("Username") },
+                modifier = Modifier.testTag("UsernameField"))
+
+            val painter =
+                rememberAsyncImagePainter(
+                    ImageRequest.Builder(context)
+                        .data(
+                            if (profilePictureUrl.isNotBlank()) profilePictureUrl
```

Comment:
Detail: You can use this pretty nice syntax: 
`profilePictureUrl.ifBlank { defaultImageUrl })`